broker/broker.go

```diff
@@ -2,53 +2,305 @@ package broker
+	if err != nil {
+		return binding, err
+	}
+	password, _ := bindingURL.User.Password()
+
+	binding.Credentials = Credentials{
+		Host:     bindingURL.Hostname(),
+		Port:     bindingURL.Port(),
+		Name:     bindingURL.RequestURI(),
+		Username: bindingURL.User.Username(),
+		Password: password,
+		URI:      bindingURL.String(),
+		JDBCURI:  JDBCURI(bindingURL.Hostname(), bindingURL.Port(), bindingURL.RequestURI(), bindingURL.User.Username(), password),
+	}
+
+	return binding, nil
 }
 
 func (b *Broker) Unbind(context context.Context, instanceID, bindingID string, details brokerapi.UnbindDetails) error {
-	return fmt.Errorf("%s", "Can't unbind an instance")
+	b.Logger.Debug("unbind", lager.Data{
+		instanceIDLogKey: instanceID,
+		bindingIDLogKey:  bindingID,
+		detailsLogKey:    details,
+	})
+
+	return nil
 }
 
 func (b *Broker) Update(context context.Context, instanceID string, details brokerapi.UpdateDetails, asyncAllowed bool) (brokerapi.UpdateServiceSpec, error) {
-	return brokerapi.UpdateServiceSpec{}, fmt.Errorf("%s", "Can't update an instance")
+	b.Logger.Debug("update", lager.Data{
+		instanceIDLogKey:   instanceID,
+		detailsLogKey:      details,
+		asyncAllowedLogKey: asyncAllowed,
+	})
+
+	spec := brokerapi.UpdateServiceSpec{}
+
+	if !asyncAllowed {
+		return spec, brokerapi.ErrAsyncRequired
+	}
+
+	service, err := b.ComposeCatalog.GetService(details.ServiceID)
+	if err != nil {
+		return spec, err
+	}
+
+	deployment, err := findDeployment(b.Compose, instanceID)
+	if err != nil {
+		return spec, err
+	}
+
+	if details.PlanID != details.PreviousValues.PlanID {
+		return spec, fmt.Errorf("changing plans is not currently supported")
+	}
+
+	plan, err := service.GetPlan(details.PlanID)
+	if err != nil {
+		return spec, err
+	}
+
+	params := composeapi.ScalingsParams{
+		DeploymentID: deployment.ID,
+		Units:        plan.Metadata.Units,
+	}
+
+	recipe, errs := b.Compose.SetScalings(params)
+	if len(errs) > 0 {
+		return spec, squashErrors(errs)
+	}
+
+	spec.IsAsync = true
+	operationData, err := getOperationData("update", recipe.ID)
+	if err != nil {
+		return spec, err
+	}
+
+	spec.OperationData = operationData
+
+	return spec, nil
 }
 
-func (b *Broker) LastOperation(context context.Context, instanceID, operationData string) (brokerapi.LastOperation, error) {
-	return brokerapi.LastOperation{}, fmt.Errorf("%s", "Can't check last operation")
+func (b *Broker) LastOperation(context context.Context, instanceID, operationDataJson string) (brokerapi.LastOperation, error) {
+
+	lastOperation := brokerapi.LastOperation{}
+	operationData := OperationData{}
+	err := json.Unmarshal([]byte(operationDataJson), &operationData)
+	if err != nil {
+		return lastOperation, err
+	}
+
+	b.Logger.Debug("last-operation", lager.Data{
+		instanceIDLogKey:    instanceID,
+		operationDataLogKey: operationData.RecipeID,
+	})
+
+	// FIXME: This is due to Compose API bug that returns 401 straight after deprovison call.
```

Comment:
Compose fixed this bug. @henrytk is going to test whether it works as expected now. If it does then @combor will remove this condition. We're going to leave the response body as JSON, rather than just the recipe string, in case we need the structure again in the future.